### PR TITLE
feat(new-menu): Allow to set the category for entries

### PR DESCRIPTION
**Pull request by susnux**
Required for the last part of: https://github.com/nextcloud/server/issues/43728

### CODE REVIEW

File: lib/newFileMenu.ts [hunk 1]
```diff
@@ -23,10 +23,32 @@
...
+	Other = 2,
+}
+
 export interface Entry {
 	/** Unique ID */
 	id: string
 
+	/**
+	 * Category to put this entry in
+	 * Added with v3.3.0 (supported since Nextcloud 30)
```
Comment (Pytal) [inline lib/newFileMenu.ts:47]: Maybe https://jsdoc.app/tags-since ?

**Review by Pytal — APPROVED** [review]

File: lib/newFileMenu.ts (hunk 1 above)
Comment (susnux) [reply lib/newFileMenu.ts:47]: I tried but it results in eslint warning :(

Comment (susnux) [reply lib/newFileMenu.ts]: Found a solution. Not that nice, but better :white_check_mark:

**Comment by skjnldsv** [comment]
🎉🎉

**Pull request merged**